app/api/templates/templates.js

```diff
@@ -49,43 +50,57 @@ const updateTranslation = (currentTemplate, template) => {
+    .then(() => {
+      if (template._id) {
+        return this._update(template, language);
+      }
+      return model.save(template)
+      .then(newTemplate => addTemplateTranslation(newTemplate)
+      .then(() => newTemplate));
+    });
+  },
+
+  _update(template, language) {
     let _currentTemplate;
-    if (template._id) {
-      return this.getById(template._id)
-      .then(currentTemplate => Promise.all([currentTemplate, updateTranslation(currentTemplate, template)]))
-      .then(([currentTemplate]) => {
-        _currentTemplate = currentTemplate;
-        currentTemplate.properties = currentTemplate.properties || [];
-        const currentTemplateContentProperties = currentTemplate.properties.filter(p => p.content);
-        const templateContentProperties = template.properties.filter(p => p.content);
-        const toRemoveValues = {};
-        currentTemplateContentProperties.forEach((prop) => {
-          const sameProperty = templateContentProperties.find(p => p.id === prop.id);
-          if (sameProperty && sameProperty.content !== prop.content) {
-            toRemoveValues[sameProperty.name] = prop.type === 'multiselect' ? [] : '';
-          }
-        });
-        if (Object.keys(toRemoveValues).length === 0) {
-          return;
+    return this.getById(template._id)
+    .then((currentTemplate) => {
+      currentTemplate.properties = currentTemplate.properties || [];
+      currentTemplate.properties.forEach((prop) => {
+        const swapingNameWithExistingProperty = template.properties.find(p => p.name === prop.name && p.id !== prop.id);
+        if (swapingNameWithExistingProperty) {
+          throw createError(`Properties can't swap names: ${prop.name}`, 400);
         }
-        return entities.removeValuesFromEntities(toRemoveValues, currentTemplate._id);
-      })
-      .then(() => save(template))
-      .then(savedTemplate => entities.updateMetadataProperties(template, _currentTemplate, language)
-      .then(() => savedTemplate));
-    }
-
-    return save(template)
-    .then(newTemplate => addTemplateTranslation(newTemplate)
-    .then(() => newTemplate));
+      });
+
+      return currentTemplate;
+    })
+    .then(currentTemplate => Promise.all([currentTemplate, updateTranslation(currentTemplate, template)]))
+    .then(([currentTemplate]) => {
+      _currentTemplate = currentTemplate;
+      const currentTemplateContentProperties = currentTemplate.properties.filter(p => p.content);
+      const templateContentProperties = template.properties.filter(p => p.content);
+      const toRemoveValues = {};
+      currentTemplateContentProperties.forEach((prop) => {
+        const sameProperty = templateContentProperties.find(p => p.id === prop.id);
+        if (sameProperty && sameProperty.content !== prop.content) {
+          toRemoveValues[sameProperty.name] = prop.type === 'multiselect' ? [] : '';
+        }
+      });
+      if (Object.keys(toRemoveValues).length === 0) {
+        return;
```

Comment:
I think that the consistent returns is a good practice, not only for eslint purposes.  If lit 98 is returning something, this line should also return something other than 'undefined'.  If removeValuesFromEntities is asynchronous, perhaps Promise.resolve() is enough.  This is much clear when you refactor functions to use async-await.